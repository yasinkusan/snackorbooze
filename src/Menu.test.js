import React from 'react';
import Menu from './FoodMenu';
import { MemoryRouter } from 'react-router-dom';

import { render, fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

function sleep(time) {
	return new Promise((resolve) => setTimeout(resolve, time));
}
const snacks = [
	{
		id          : 'nachos',
		name        : 'Nachos',
		description : 'An American classic!',
		recipe      : 'Cover expensive, organic tortilla chips with Cheez Whiz.',
		serve       : 'Serve in a hand-thrown ceramic bowl, garnished with canned black olives'
	},
	{
		id          : 'hummus',
		name        : 'Hummus',
		description : 'Sure to impress your vegan friends!',
		recipe      : 'Purchase one container of hummus.',
		serve       : 'Place unceremoniously on the table, along with pita bread.'
	},
	{
		id          : 'arugula-and-walnut-salad',
		name        : 'Arugula and Walnut Salad',
		description : 'Tart and delicious.',
		recipe      :
			'Mix arugula, toasted walnuts, and thinly-sliced Parmesan cheese. Dress with lemon and olive oil.',
		serve       : 'Place on tiny, precious little plates.'
	}
];

it('renders without crashing', () => {
	render(
		<MemoryRouter initialEntries={[ '/snacks' ]}>
			<Menu items={snacks} title="Snacks" remove={jest.fn()} />
		</MemoryRouter>
	);
});

it('matches loading snapshot', () => {
	const { asFragment } = render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<Menu items={snacks} title="Snacks" remove={jest.fn()} />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
it('matches rendered snapshot', async () => {
	const { asFragment } = render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<Menu items={snacks} title="Snacks" remove={jest.fn()} />
		</MemoryRouter>
	);
	await act(() => sleep(1100));
	expect(asFragment()).toMatchSnapshot();
});
