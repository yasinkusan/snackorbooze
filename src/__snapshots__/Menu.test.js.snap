// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches loading snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Snacks Menu
        </h5>
        <p
          class="card-text"
          style="text-align: center;"
        >
          Explore all of our delish snacks below!
        </p>
        <ul
          class="list-group"
        >
          <div
            class="Item"
          >
            <a
              data-testid="link-nachos"
              href="/snacks/nachos"
            >
              <li
                class="list-group-item"
              >
                Nachos
              </li>
            </a>
          </div>
          <div
            class="Item"
          >
            <a
              data-testid="link-hummus"
              href="/snacks/hummus"
            >
              <li
                class="list-group-item"
              >
                Hummus
              </li>
            </a>
          </div>
          <div
            class="Item"
          >
            <a
              data-testid="link-arugula-and-walnut-salad"
              href="/snacks/arugula-and-walnut-salad"
            >
              <li
                class="list-group-item"
              >
                Arugula and Walnut Salad
              </li>
            </a>
          </div>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`matches rendered snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Snacks Menu
        </h5>
        <p
          class="card-text"
          style="text-align: center;"
        >
          Explore all of our delish snacks below!
        </p>
        <ul
          class="list-group"
        >
          <div
            class="Item"
          >
            <a
              data-testid="link-nachos"
              href="/snacks/nachos"
            >
              <li
                class="list-group-item"
              >
                Nachos
              </li>
            </a>
          </div>
          <div
            class="Item"
          >
            <a
              data-testid="link-hummus"
              href="/snacks/hummus"
            >
              <li
                class="list-group-item"
              >
                Hummus
              </li>
            </a>
          </div>
          <div
            class="Item"
          >
            <a
              data-testid="link-arugula-and-walnut-salad"
              href="/snacks/arugula-and-walnut-salad"
            >
              <li
                class="list-group-item"
              >
                Arugula and Walnut Salad
              </li>
            </a>
          </div>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
